test/run.d

```diff
@@ -121,6 +121,7 @@ Options:
         auto taskPool = new TaskPool(jobs);
         scope(exit) taskPool.finish();
         ensureToolsExists(EnumMembers!TestTools);
+        buildDShellPrebuilt(env);
```

Comment:
I could do that but the dshell_prebuilt object file is built by the `d_do_test` tool and therefore depends on it.  I initially tried having `run.d` build it rather than `d_do_test` but couldn't get it to work when I realized that having `d_do_test` build it would make more sense as it already has the right flags and environment because it's what actually builds the dshell tests.

The simplest way I can think to do that at the moment is add this to the end of ensureToolsExists

```
if (tool.name == "d_do_test")
{
    // now build dshell_prebuilt
    const testRunnerExe = resultsDir.buildPath(TestTools.testRunner.name).exeName;
    const command = [testRunnerExe, "-dshell_prebuilt"];
    spawnProcess(command, env, Config.none).wait;
}

It would also mean adding `env` as another argument to ensureToolsExist because it's required to know what flags to use when building the dshell_prebuilt object.

So long as you're ok with these things I'll change it.  Or if you have a better idea let me know.

Reply:
Nevermind, got `run.d` and `Makefile` to build it.  I think I just wasn't passing the whole environment.